packages/firestore/src/core/sync_engine.ts

```diff
@@ -154,6 +155,9 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
   private limboResolutionsByTarget: {
     [targetId: number]: LimboResolution;
   } = {};
+  private readonly maxConcurrentLimboResolutions: number = DEFAULT_MAX_CONCURRENT_LIMBO_RESOLUTIONS;
+  /** The keys of documents whose limbo resolutions are enqueued. */
+  private limboListenQueue: DocumentKey[] = [];
```

Comment:
I'm still concerned about the "listen" in `limboListenQueue` as a name here. Even your comment doesn't mention "listens" so it's not clear if there's an actual conceptual difference between limbo resolution and a limbo listen or if this is just an accidental inconsistency that wasn't caught in review.

Since you've resolved to distinguish active and enqueued limbo resolutions, why can't this be `enqueuedLimboResolutions`? That is: instead of focusing on the fact that they're enqueued to start a listener, just know that these are limbo resolutions in the enqueued state without specifying exactly what they're waiting for. This resolves the inconsistency without any effective loss of clarity, because exactly what's next after enqueueing isn't a critical detail in this name.

If you prefer "Queue" in the name, "deferred" might better name for the enqueued resolutions because then you can make this `deferredLimboResolutionQueue`. `enqueuedLimboResolutionQueue` looks strange.

Reply:
Done.  I've renamed the variable to `enqueuedLimboResolutions`.